File: DotNetGotchas/CSharp/Finalize/WhyNoFinalize/SomeClass.cs
using System;

namespace WhyNoFinalize
{
	public class SomeClass
	{
		private SomeOtherClass ref1;
		
		public SomeClass(SomeOtherClass givenObject)
		{
			ref1 = givenObject;
		}

		~SomeClass()
		{
			ref1 = null;
		}
	}
}

File: DotNetGotchas/VBDN/Finalize/WhyNoFinalize/SomeClass.vb
Public Class SomeClass
    Private ref1 As SomeOtherClass

    Public Sub New(ByVal givenObject As SomeOtherClass)
        ref1 = givenObject
    End Sub
    Protected Overrides Sub Finalize()
        ref1 = Nothing
        MyBase.Finalize()
    End Sub
End Class
